### PR TITLE
doc: Improve `README.md`

### PR DESCRIPTION
**Pull request by Hywan**
# Description

This PR improves our `README.md`.

* The “main” one has a better description of what Wasmer is by giving the keywords in the first sentence. It also lists more way to install Wasmer CLI. It removes the use of `<quote />` to indicate a note, it's semantically invalid. It explains super quickly the notion of compilers and engines by using the words “compilation strategies” and “artifact strategies”. It fixes various links. And more.
* It adds a `README.md` to the following directories:
  * `assets/`
  * `benches/`
   * `docs/`
  * `lib/`
* It improves the `README.md` of the following directories:
  * `fuzz/`

### CODE REVIEW

**Review by jubianchi — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -27,56 +27,143 @@
...
+  fully sandboxed environment.
 
-* **Pluggable**. Wasmer supports different compilation frameworks to best suit your needs (LLVM, Cranelift...).
+* **Pluggable**. To best suit your needs, Wasmer supports different
+  compilation strategies (_aka_ the compilers — based on LLVM, based
+  on Cranelift, or Singlepass) and artifact strategies (_aka_ the
+  engines — Universal, Dylib, Staticlib).
 
-* **Universal**. You can run Wasmer in any *platform* (macOS, Linux and Windows) and *chipset*.
+* **Universal**. You can run Wasmer in any _platform_ (Linux, macOS
```
[the remark this answers is not in the document]
Comment (Hywan) [reply README.md:52]: :+1: thanks!

File: README.md [hunk 2]
```diff
@@ -27,56 +27,143 @@
...
+    brew install wasmer
+    ```
+
+  </details>
+
+* <details>
+    <summary>With <a href="https://github.com/ScoopInstaller/Main/blob/master/bucket/wasmer.json">Scoop</a></summary>
+
+    ```sh
+    scopp install wasmer
```
[the remark this answers is not in the document]
Comment (Hywan) [reply README.md]: :+1: thanks! good eyes!

**Comment by Hywan** [comment]
If everyone agrees on the new `/README.md`, we should then update the translations.

File: SECURITY.md [hunk 3]
```diff
@@ -2,18 +2,20 @@
...
-While in beta, the latest published version of `wasmer` (`0.x`) will be supported with security updates.
+The table below summarizes which versions are still supported, and which aren't.
 
-| Version | Supported          |
-| ------- | ------------------ |
-| 0.x   | :white_check_mark: |
+| Version | Supported |
+|-|-|
+| 0.x | ❌ |
+| 1.x | ✅ |
```
Comment (jubianchi) [inline SECURITY.md:10]: We should add the 2.x here I think

**Review by jubianchi — APPROVED** [review]

File: SECURITY.md (hunk 3 above)
Comment (Hywan) [reply SECURITY.md:10]: Not yet. It's not released :-).

**Comment by syrusakbary** [comment]
I ended refactoring again the main README. It was optimized for readability and being concise in propose.
However, there are a few things that you did improve in the right direction. So I just went a step forward on it!

Note: we did use `<quote>` because how it renders on Github, not because it was semantically valid.

Here's before: https://github.com/wasmerio/wasmer/tree/5f46b9dd260fe1c6fd1c7d46b0d887daacc0e6c2
Here's after: https://github.com/wasmerio/wasmer/tree/d2e22d6ea572e4a76813ede70a69119dfa0aaa79

**Pull request merged**